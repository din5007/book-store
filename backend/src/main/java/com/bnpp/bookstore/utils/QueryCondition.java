package com.bnpp.bookstore.utils;

import io.micrometer.common.util.StringUtils;
import org.springframework.stereotype.Component;

@Component("queryUtil")
public class QueryCondition {

  public boolean checkNull(String field) {
    return StringUtils.isBlank(field);
  }
}
