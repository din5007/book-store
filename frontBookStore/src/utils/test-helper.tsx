import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";


function customRender(component : any , store?: any) {

    return render(<StoreProvider store={store}>
        <BrowserRouter>
            {component}
        </BrowserRouter>
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;