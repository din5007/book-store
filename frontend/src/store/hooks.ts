import { createTypedHooks } from "easy-peasy";

import { AppStoreModel } from ".";

export const { useStoreActions, useStoreState } = createTypedHooks<AppStoreModel>();