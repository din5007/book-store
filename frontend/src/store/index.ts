import { HttpService } from "../service/http/http.service";
import { createStore } from 'easy-peasy';
import { BookModel, booksModel } from "./books/books.model";
import { cartModel, CartModel } from "./cart/cart.model";
import { userModel, UserModel } from "./users/user.model";
import { OrderModel, orderModel } from "./order/order.model";

export interface AppStoreModel {
    books: BookModel;
    cart: CartModel;
    user: UserModel;
    order: OrderModel;
}

export const appStoreModel : AppStoreModel = {
    books: booksModel,
    cart: cartModel,
    user: userModel,
    order: orderModel,
}

export function storeModel() {
    return createStore(appStoreModel, {
        injections: {
            httpService: new HttpService()
        },
        initialState: {
            
        },
    })
}
export const store =  storeModel();