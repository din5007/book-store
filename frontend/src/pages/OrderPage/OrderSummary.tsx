import { Button, ListGroup } from "react-bootstrap";
import { useStoreActions, useStoreState } from "../../store/hooks";

const OrderSummary = () => {
  const { cart, totalPrice } = useStoreState((store) => store.cart);
  const { createOrder } = useStoreActions((actions) => actions.order);



  return (
    <div style={{ margin : '5%'}}>
      <div className="page d-flex justify-content-between">
        <h1> Your Order Summary </h1>
        <Button
          variant="primary"
          size="lg"
          disabled={!cart.length}
          onClick={() => createOrder({})}
        >
          Order Now !
        </Button>
      </div>
      <div style={{ margin : '50px'}}>
        <ListGroup>
            {cart?.map(({ book, quantity }) => (
            <ListGroup.Item as="li">
                <strong>{book?.title.toUpperCase()}</strong> <span style={{ float : 'right'}}>{quantity} * {book.price} = {quantity * book.price}</span>
            </ListGroup.Item>
            ))}
        </ListGroup>
        <h3 style={{ float: "right"}}> Final Order Value : {totalPrice}</h3>
      </div>
    </div>
  );
};

export default OrderSummary;
