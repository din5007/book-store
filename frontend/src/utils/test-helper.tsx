import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { AppStoreModel, store } from "../store";
import { Router } from 'react-router-dom';
import { ReactNode } from "react";



// const provider : React.FC<{children : ReactNode, store: any}> = ({ children, store }) => {
//     return (
        
//     )
// }

function customRender(component : any , store: any) {

    return render(<StoreProvider store={store}>
        {component}
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;