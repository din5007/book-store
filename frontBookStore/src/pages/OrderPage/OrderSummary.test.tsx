import { createStore } from 'easy-peasy';
import { appStoreModel } from '../../store';
import { customRender as render } from '../../utils/test-helper';
import OrderSummary from "./OrderSummary"
import { describe, expect, it, vi } from 'vitest';

const mock = {
    get: vi.fn()
};

const store = createStore(appStoreModel, {
    injections: {
        httpService : mock
    },
    initialState: {
        order : {
            
        }
    }
})

describe('Order Summary Page', () => {
    it('Order Summary Page should load properly - snapshot', async () => {
        const {container}  = render(<OrderSummary />, store);
        expect(container).toMatchSnapshot();
    });
})