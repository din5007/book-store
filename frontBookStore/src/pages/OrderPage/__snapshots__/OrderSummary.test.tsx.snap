// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Order Summary Page > Order Summary Page should load properly - snapshot 1`] = `
<div>
  <div
    style="margin: 5%;"
  >
    <div
      class="page d-flex justify-content-between"
    >
      <h1>
         Your Order Summary 
      </h1>
      <button
        class="btn btn-primary btn-lg"
        disabled=""
        type="button"
      >
        Order Now !
      </button>
    </div>
    <div
      style="margin: 50px;"
    >
      <div
        class="list-group"
      />
      <h3
        style="css-float: right;"
      >
         Final Order Value : 
        0
      </h3>
    </div>
  </div>
</div>
`;
